velox/exec/HashBuild.cpp

```diff
@@ -1125,15 +1130,14 @@ void HashBuild::reclaim(
     HashBuild* buildOp = dynamic_cast<HashBuild*>(op);
     VELOX_CHECK_NOT_NULL(buildOp);
     VELOX_CHECK(buildOp->canReclaim());
-    if ((buildOp->state_ != State::kRunning &&
-         buildOp->state_ != State::kWaitForBuild) ||
-        buildOp->nonReclaimableSection_) {
+    if (buildOp->nonReclaimableState()) {
       // TODO: reduce the log frequency if it is too verbose.
       ++stats.numNonReclaimableAttempts;
       LOG(WARNING) << "Can't reclaim from hash build operator, state_["
                    << stateName(buildOp->state_) << "], nonReclaimableSection_["
-                   << buildOp->nonReclaimableSection_ << "], "
-                   << buildOp->pool()->name();
+                   << nonReclaimableSection_ << "], spiller_["
+                   << (spiller_->finalized() ? "finalized" : "non-finalized")
+                   << "], " << buildOp->pool()->name();
```

Comment:
The logging message is duplicated maybe we can extract them out into a single method or something?